app/views/spree/checkout/payment/_affirm.html.erb

```diff
@@ -1,8 +1,7 @@
 <%= javascript_include_tag "spree/frontend/affirm_checkout" %>
 <%
   config = {
-    # confirmation_url: confirm_affirm_url(payment_method_id: payment_method.id),
-    confirmation_url: "http://localhost:3000",
+    confirmation_url: confirm_affirm_url(payment_method_id: payment_method.id),
```

Comment:
It would probably be best to do the frontend side of things last, instead of defining the callbacks controller after the frontend. As the callbacks controller is a dependency on making this work on the frontend.